src/concepts/kafka-proxies/http-proxy.md

```diff
@@ -0,0 +1,67 @@
+---
+description: Zilla lets you configure application-centric REST API and SSE stream endpoints that unlock Kafka event-driven architectures.
+prev: false
+next: /tutorials/rest/rest-intro.md
+---
+
+# HTTP Kafka Proxy
+
+Zilla lets you configure application-centric REST API and SSE stream endpoints that unlock Kafka event-driven architectures. Zilla gives the developer freedom to define their own HTTP mapping to Kafka, with control over the topics, message key, message headers, and payload. Zilla enables an HTTP client to connect and interact with Kafka without needing to understand Kafka-specific paradigms.
+
+## Configure Endpoints
+
+Zilla can be configured to map REST APIs to Kafka using the [http-kafka](../../reference/config/bindings/binding-http-kafka.md) binding in `zilla.yaml`. Zilla can map route REST urls using [wildcard pattern matching](../../concepts/config-intro.md#pattern-matching) and [dynamic path params](../../concepts/config-intro.md#dynamic-path-parameters).
+
+### HTTP request methods
+
+Zilla separates the HTTP request methods in two groups called capabilities, produce and fetch. The [produce](../../concepts/config-intro.md#the-fetch-capability) capability handles method types `POST`, `PUT`, `DELETE`, and `PATCH` that put messages onto Kafka topics. The [fetch](../../reference/config/bindings/binding-http-kafka.md#with-capability-fetch) capability handles the `GET` method that gets messages off of Kafka topics.
+
+## Correlated Request-Response
+
+Zilla manages the HTTP lifecycle with the request and response payload on the event stream. Each message is correlated to each other with a `zilla:correlation-id` header, providing an identifier for both Zilla and Kafka workflows to act on. Correlated messages can be on the same or different Kafka topics.
+
+### sync
+
+A synchronous interaction from the client will open the connection and wait for the correlated response message to be delivered to the caller.
+
+### async
+
+An asynchronous interaction is managed over a pair of Kafka topics. An initiating request can include a `prefer: respond-async` header which will immediately return with `202 Accepted` plus the location to retrieve a correlated response. The client then sends a request to the returned location with the `prefer: wait=N` header to retrieve the correlated response as soon as it becomes available, removing the need for client polling.
```

Comment:
Can we handle both situations with a pair of messages on a single topic but two separate topics is a better architecture? Or is that a gRPC thing.

Reply:
It has to be two topics because messages flow in one direction only on a topic so we need to have a separate request topic and response topic. 

This is a generally accepted design pattern for request response over messaging. 

https://www.enterpriseintegrationpatterns.com/patterns/messaging/RequestReply.html